django/db/backends/base/base.py

```diff
@@ -633,6 +633,8 @@ def schema_editor(self, *args, **kwargs):
 
     def on_commit(self, func):
         if self.in_atomic_block:
+            if not callable(func):
+                raise TypeError("on_commit()'s callback must be a callable.")
```

Comment:
`func` is called even if no transaction is in progress, so we should move this to the first line. Fixed.